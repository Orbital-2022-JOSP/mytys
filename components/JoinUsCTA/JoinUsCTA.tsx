import { GenericContainer } from '../GenericContainer/GenericContainer';

export const JoinUsCTA: React.FC = () => {
    return (
        <GenericContainer>
            <h1 className="text-5xl text-light-blue-500 text-center pb-7 font-medium"> Join Us </h1>
            <div className="grid max-w-screen-lg gap-8 row-gap-5 md:row-gap-8 sm:mx-auto lg:grid-cols-2">
                <div className="transition duration-300 transform bg-white rounded shadow-sm hover:-translate-y-1 hover:shadow md:text-center">
                    <div className="relative">
                        <img
                            className="object-cover w-full h-64 rounded-t lg:h-80 xl:h-96"
                            src="images/mytys_landing_bg_3.jpg"
                            alt=""
                        />
                        <div className="absolute inset-0 bg-gray-800 bg-opacity-25" />
                    </div>
                    <div className="px-6 py-8 border border-t-0 rounded-b sm:px-8">
                        <h5 className="mb-2 text-xl font-bold leading-none sm:text-2xl">
                            As a Student
                        </h5>
                        <p className="mb-5 text-gray-700">
                            We provide an online-TYS style form of gamified education, proven to cultivate joy in learning and improve results in our students
                        </p>
                        <button
                            type="submit"
                            className="inline-flex items-center justify-center h-12 px-6 font-medium tracking-wide text-white transition duration-200 rounded shadow-md bg-light-blue-accent-400 hover:bg-light-blue-accent-700 focus:shadow-outline focus:outline-none"
                        >
                            Enroll Now
                        </button>
                    </div>
                </div>
                <div className="transition duration-300 transform bg-white rounded shadow-sm hover:-translate-y-1 hover:shadow md:text-center">
                    <div className="relative">
                        <img
                            className="object-cover w-full h-64 rounded-t lg:h-80 xl:h-96"
                            src="images/Teacher1.jpg"
                            alt=""
                        />
                        <div className="absolute inset-0 bg-gray-800 bg-opacity-25" />
                    </div>
                    <div className="px-6 py-8 border border-t-0 rounded-b sm:px-8">
                        <h5 className="mb-2 text-xl font-bold leading-none sm:text-2xl">
                            As a Tutor
                        </h5>
                        <p className="mb-5 text-gray-700">
                            As a tutor at MYTYS, you would play a key role in helping students achieve their academic results and in improving their quality of education
                        </p>
                        <button
                            type="submit"
                            className="inline-flex items-center justify-center h-12 px-6 font-medium tracking-wide text-white transition duration-200 rounded shadow-md bg-light-blue-accent-400 hover:bg-light-blue-accent-700 focus:shadow-outline focus:outline-none"
                        >
                            Join us
                        </button>
                    </div>
                </div>
            </div>
        </GenericContainer>
    );
};