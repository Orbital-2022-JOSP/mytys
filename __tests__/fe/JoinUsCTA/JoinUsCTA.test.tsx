import { render } from '@testing-library/react';
import { JoinUsCTA } from '../../../components/JoinUsCTA/JoinUsCTA';

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {};
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'unauthenticated' }
        }),
    };
});

describe("The Join Us CTA Component", () => {

    it('Should render correctly',
        async () => {
            const { container } = render(<JoinUsCTA />);

            expect(container).toMatchSnapshot()
        })
})

