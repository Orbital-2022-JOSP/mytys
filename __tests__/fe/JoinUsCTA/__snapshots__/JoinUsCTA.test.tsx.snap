// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Join Us CTA Component Should render correctly 1`] = `
<div>
  <div
    class="px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl sm:min-w-xl md:min-w-full lg:min-w-screen-lg xl:min-w-screen-xl md:px-24 lg:px-8 lg:py-20"
  >
    <h1
      class="text-5xl text-light-blue-500 text-center pb-7 font-medium"
    >
       Join Us 
    </h1>
    <div
      class="grid max-w-screen-lg gap-8 row-gap-5 md:row-gap-8 sm:mx-auto lg:grid-cols-2"
    >
      <div
        class="transition duration-300 transform bg-white rounded shadow-sm hover:-translate-y-1 hover:shadow md:text-center"
      >
        <div
          class="relative"
        >
          <img
            alt=""
            class="object-cover w-full h-64 rounded-t lg:h-80 xl:h-96"
            src="images/mytys_landing_bg_3.jpg"
          />
          <div
            class="absolute inset-0 bg-gray-800 bg-opacity-25"
          />
        </div>
        <div
          class="px-6 py-8 border border-t-0 rounded-b sm:px-8"
        >
          <h5
            class="mb-2 text-xl font-bold leading-none sm:text-2xl"
          >
            As a Student
          </h5>
          <p
            class="mb-5 text-gray-700"
          >
            We provide an online-TYS style form of gamified education, proven to cultivate joy in learning and improve results in our students
          </p>
          <button
            class="inline-flex items-center justify-center h-12 px-6 font-medium tracking-wide text-white transition duration-200 rounded shadow-md bg-light-blue-accent-400 hover:bg-light-blue-accent-700 focus:shadow-outline focus:outline-none"
            type="submit"
          >
            Enroll Now
          </button>
        </div>
      </div>
      <div
        class="transition duration-300 transform bg-white rounded shadow-sm hover:-translate-y-1 hover:shadow md:text-center"
      >
        <div
          class="relative"
        >
          <img
            alt=""
            class="object-cover w-full h-64 rounded-t lg:h-80 xl:h-96"
            src="images/Teacher1.jpg"
          />
          <div
            class="absolute inset-0 bg-gray-800 bg-opacity-25"
          />
        </div>
        <div
          class="px-6 py-8 border border-t-0 rounded-b sm:px-8"
        >
          <h5
            class="mb-2 text-xl font-bold leading-none sm:text-2xl"
          >
            As a Tutor
          </h5>
          <p
            class="mb-5 text-gray-700"
          >
            As a tutor at MYTYS, you would play a key role in helping students achieve their academic results and in improving their quality of education
          </p>
          <button
            class="inline-flex items-center justify-center h-12 px-6 font-medium tracking-wide text-white transition duration-200 rounded shadow-md bg-light-blue-accent-400 hover:bg-light-blue-accent-700 focus:shadow-outline focus:outline-none"
            type="submit"
          >
            Join us
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
